dotCMS/src/main/java/com/dotcms/vanityurl/model/CachedVanityUrl.java

```diff
@@ -45,113 +53,40 @@ public CachedVanityUrl(final VanityUrl vanityUrl) {
-        this.languageId = fromCachedVanityUrl.getLanguageId();
-        this.siteId = fromCachedVanityUrl.getSiteId();
-        this.forwardTo = fromCachedVanityUrl.getForwardTo();
-        this.response = fromCachedVanityUrl.getResponse();
-        this.order    = fromCachedVanityUrl.getOrder();
-    }
-
-    /**
-     * Generates a CachedVanityUrl from another given CachedVanityUrl
-     *
-     * @param forwardTo replace the forward.
-     * @param fromCachedVanityUrl VanityURL to copy
-     *
-     */
-    public CachedVanityUrl(final String forwardTo,
-                           final CachedVanityUrl fromCachedVanityUrl) {
-
-
-        this.pattern     = fromCachedVanityUrl.pattern;
-        this.vanityUrlId = fromCachedVanityUrl.getVanityUrlId();
-        this.url         = fromCachedVanityUrl.url;
-        this.languageId  = fromCachedVanityUrl.getLanguageId();
-        this.siteId      = fromCachedVanityUrl.getSiteId();
-        this.forwardTo   = forwardTo;
-        this.response    = fromCachedVanityUrl.getResponse();
-        this.order       = fromCachedVanityUrl.getOrder();
-    }
-
-    public int getOrder() {
-        return order;
-    }
-
-    /**
-     * Get the URL from the Cached Vanity URL
-     *
-     * @return the URL from the Cached Vanity URL
-     */
-    public String getUrl() {
-        return url;
-    }
-
-    /**
-     * Get the Site Id from the Cached Vanity URL
-     *
-     * @return the Site ID from the Cached Vanity URL
-     */
-    public String getSiteId() {
-        return siteId;
-    }
-
-    /**
-     * Get the Language Id from the Cached Vanity URL
-     *
-     * @return the language Id from the Cached Vanity URL
-     */
-    public long getLanguageId() {
-        return languageId;
+        this.languageId = languageId;
+        this.siteId = siteId;
+        this.forwardTo = forwardTo;
+        this.response = response;
+        this.order    = 0;
     }
-
-    /**
-     * Get the Forward to path from the Cached Vanity URL
-     *
-     * @return the Forward to path from the Cached Vanity URL
-     */
-    public String getForwardTo() {
-        return forwardTo;
-    }
-
-    /**
-     * Get the Response code from the Cached Vanity URL
-     *
-     * @return the Response code from the Cached Vanity URL
-     */
-    public int getResponse() {
-        return response;
-    }
-
-    /**
-     * Get the URI Pattern from the Cached Vanity URL
-     *
-     * @return the URI Pattern from the Cached Vanity URL
-     */
-    public Pattern getPattern() {
-        return pattern;
-    }
-
-    /**
-     * get the Vanitu Url Identifier
-     *
-     * @return The Vanity Url Identifier
-     */
-    public String getVanityUrlId() {
-        return vanityUrlId;
+    
+    public Tuple2<String, String> processForward(final String url) {
+      String newForward = this.forwardTo;
+      String queryString = null;
+      if(pattern!=null) {
+        Matcher matcher = pattern.matcher(url);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'matcher' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)